packages/devtools_app/lib/src/screens/inspector_v2/widget_details.dart

```diff
@@ -35,44 +38,78 @@ class _WidgetDetailsState extends State<WidgetDetails> with AutoDisposeMixin {
 
   @override
   Widget build(BuildContext context) {
-    return ValueListenableBuilder<InspectorTreeNode?>(
-      valueListenable: controller.selectedNode,
-      builder: (context, _, __) {
-        final node = selectedNode;
-        if (node == null) {
-          return const Center(
-            child: Text(
-              'Select a widget to view its layout.',
-              textAlign: TextAlign.center,
-              overflow: TextOverflow.clip,
-            ),
-          );
-        }
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        final parentHeight = constraints.maxHeight;
+        final parentWidth = constraints.maxWidth;
+        final horizontalLayout = parentWidth >= (parentHeight * 1.25);
+        final canFitLayoutExplorerVertically =
+            parentHeight >= WidgetDetails.layoutExplorerHeight;
+        final canFitLayoutExplorerHorizontally =
+            parentWidth >= WidgetDetails.layoutExplorerWidth;
+        final canFitLayoutExplorer =
+            canFitLayoutExplorerVertically && canFitLayoutExplorerHorizontally;
+        final canFitLargeDetailsTable = horizontalLayout
+            ? parentWidth > WidgetDetails.layoutExplorerWidth * 3
+            : parentHeight > WidgetDetails.layoutExplorerHeight * 3;
 
-        return SplitPane(
-          axis: isScreenWiderThan(
-            context,
-            InspectorScreenBodyState.minScreenWidthForTextBeforeScaling,
-          )
-              ? Axis.horizontal
-              : Axis.vertical,
-          initialFractions: const [0.5, 0.5],
-          children: [
-            if (FlexLayoutExplorerWidget.shouldDisplay(node)) ...[
-              FlexLayoutExplorerWidget(controller),
-            ] else if (BoxLayoutExplorerWidget.shouldDisplay(node)) ...[
-              BoxLayoutExplorerWidget(controller),
-            ] else ...[
-              const Center(
+        return ValueListenableBuilder<InspectorTreeNode?>(
+          valueListenable: controller.selectedNode,
+          builder: (context, _, __) {
+            final node = selectedNode;
+            if (node == null) {
+              return const Center(
                 child: Text(
-                  'Currently, the Layout Explorer only supports Box and Flex-based widgets.',
+                  'Select a widget to view its layout.',
                   textAlign: TextAlign.center,
                   overflow: TextOverflow.clip,
                 ),
-              ),
-            ],
-            PropertiesView(controller: controller, node: node),
-          ],
+              );
+            }
+
+            return Flex(
+              direction: horizontalLayout ? Axis.horizontal : Axis.vertical,
+              children: [
+                if (BoxLayoutExplorerWidget.shouldDisplay(node) &&
+                    canFitLayoutExplorer) ...[
```

Comment:
I don't think you need this list spread if there's only a single widget under the conditional statement.

Reply:
The layout explorer has been moved, so this code is now gone.